DSharpPlus/Net/Rest/RateLimitPolicy.cs

```diff
@@ -121,19 +123,22 @@ public RateLimitPolicy(ILogger logger)
 
         HttpResponseMessage response = await action(context, cancellationToken);
 
-        hash = response.Headers.GetValues("X-RateLimit-Bucket").SingleOrDefault() ?? "UNKNOWN";
-
-        if(!RateLimitBucket.TryExtractRateLimitBucket(response.Headers, out RateLimitBucket? extracted))
+        if (!exemptFromGlobalLimit)
         {
-            return response;
-        }
+            bool hasBucketHeader = response.Headers.TryGetValues("X-RateLimit-Bucket", out IEnumerable<string>? hashHeader);
+            hash = hasBucketHeader ? hashHeader?.FirstOrDefault() : "UNKNOWN";
```

Comment:
if we don't have a header, dont try to add it to the known hashes. this should just be a `.Single()` call after establishing it exists

Reply:
should be solved